src/components/MainPageModel.ts

```diff
@@ -0,0 +1,282 @@
+        if (this.filter.categories.includes(category)) {
+            this.filter.categories = this.filter.categories.filter((cur) => cur !== category);
+        } else {
+            this.filter.categories.push(category);
+        }
+        this.onChangeModel(this.products, this.filter, 0, 0);
+        setQueryParameters(this.filter);
+    }
+
+    addRemoveBrand(brand: string) {
+        if (this.filter.brands.includes(brand)) {
+            this.filter.brands = this.filter.brands.filter((cur) => cur !== brand);
+        } else {
+            this.filter.brands.push(brand);
+        }
+        this.onChangeModel(this.products, this.filter, 0, 0);
+        setQueryParameters(this.filter);
+    }
+
+    addSearch(searchString: string) {
+        this.filter.search = searchString;
+        this.onChangeModel(this.products, this.filter, 0, 0);
+        setQueryParameters(this.filter);
+    }
+
+    addSort(sortString: string) {
+        this.filter.sort = sortString;
+        this.onChangeModel(this.products, this.filter, 0, 0);
+        setQueryParameters(this.filter);
+    }
+
+    addSetView(stringView: string) {
+        this.filter.view = stringView;
+        this.onChangeModel(this.products, this.filter, 0, 0);
+        setQueryParameters(this.filter);
+    }
+
+    changeMinMaxPrice(priceOne: number, priceTwo: number) {
+        if (priceOne <= priceTwo) {
+            this.filter.minPrice = priceOne;
+            this.filter.maxPrice = priceTwo;
+        } else {
+            this.filter.minPrice = priceTwo;
+            this.filter.maxPrice = priceOne;
+        }
+        this.onChangeModel(this.products, this.filter, 0, 0);
+        setQueryParameters(this.filter);
+    }
+
+    changeMinMaxStock(stockOne: number, stockTwo: number) {
+        if (stockOne <= stockTwo) {
+            this.filter.minStock = stockOne;
+            this.filter.maxStock = stockTwo;
+        } else {
+            this.filter.minStock = stockTwo;
+            this.filter.maxStock = stockOne;
+        }
+        this.onChangeModel(this.products, this.filter, 0, 0);
+        setQueryParameters(this.filter);
+    }
+
+    filterByCategory(products: IProduct[], categories: string[]) {
+        if (categories.length === 0) return products;
+        const productsFiltered: IProduct[] = [];
+        this.products.map((cur) => {
+            for (const currentCategory of categories) {
+                if (cur.category === currentCategory) {
+                    productsFiltered.push(cur);
+                }
+            }
+        });
+        return productsFiltered;
+    }
+
+    filterByBrand(products: IProduct[], brands: string[]) {
+        if (brands.length === 0) return products;
+        const productsFiltered: IProduct[] = [];
+        this.products.map((cur) => {
+            for (const currentBrand of brands) {
+                if (cur.brand === currentBrand) {
+                    productsFiltered.push(cur);
+                }
+            }
+        });
```

Comment:
Use `filter` method here instead of `map`

Reply:
Done -> a1d44dc8e64c1e3a3e3ac3f9a53d724b8aebb41f